File: backend/list_tables.py
import re

with open('../database_dump.sql', 'r', encoding='utf-8', errors='ignore') as f:
    tables = []
    for line in f:
        match = re.search(r'CREATE TABLE public\."?(\w+)"?', line)
        if match:
            tables.append(match.group(1))

print("Found tables:", sorted(list(set(tables))))

File: backend/list_tables.js
const fs = require('fs');
const readline = require('readline');

async function listTables() {
    const fileStream = fs.createReadStream('../database_dump.sql');
    const rl = readline.createInterface({
        input: fileStream,
        crlfDelay: Infinity
    });

    const tables = new Set();
    for await (const line of rl) {
        const match = line.match(/CREATE TABLE public\."?(\w+)"?/);
        if (match) {
            tables.add(match[1]);
        }
    }
    const sortedTables = Array.from(tables).sort();
    console.log("Found " + sortedTables.length + " tables:");
    sortedTables.forEach(t => console.log("- " + t));
}

listTables();
